packages/components/src/popover/index.js

```diff
@@ -417,79 +409,55 @@ const Popover = ( {
 		__unstableBoundaryParent,
 	] );
 
-	const focusOnMountRef = useFocusOnMount( focusOnMount );
-	const focusOutsideProps = useFocusOutside( handleOnFocusOutside );
-
-	// Event handlers
-	const maybeClose = ( event ) => {
-		// Close on escape
-		if ( event.keyCode === ESCAPE && onClose ) {
-			event.stopPropagation();
+	const onDialogClose = ( type, event ) => {
+		// Ideally the popover should have just a single onClose prop and
+		// not three props that potentially do the same thing.
+		if ( onClose ) {
 			onClose();
-		}
-
-		// Preserve original content prop behavior
-		if ( onKeyDown ) {
-			onKeyDown( event );
-		}
-	};
-
-	/**
-	 * Shims an onFocusOutside callback to be compatible with a deprecated
-	 * onClickOutside prop function, if provided.
-	 *
-	 * @param {FocusEvent} event Focus event from onFocusOutside.
-	 */
-	function handleOnFocusOutside( event ) {
-		// Defer to given `onFocusOutside` if specified. Call `onClose` only if
-		// both `onFocusOutside` and `onClickOutside` are unspecified. Doing so
-		// assures backwards-compatibility for prior `onClickOutside` default.
-		if ( onFocusOutside ) {
+		} else if ( type === 'focusoutside' && onFocusOutside ) {
 			onFocusOutside( event );
-			return;
-		} else if ( ! onClickOutside ) {
-			if ( onClose ) {
-				onClose();
+		} else if ( type === 'focusoutside' && onClickOutside ) {
```

Comment:
Do you think we can somehow removing this now? How long has it been deprecated?

Reply:
That's a good question, I don't really know but I think at some point we should start having some regular schedule. If something have been reprecated in the WP version v - 10 (just example), remove it. Something we might want to discuss in #core-js cc @gziolo

Reply:
Right. Especially for code that hard to maintain like this. I'm not sure if this code is even tested anywhere?